code/game/objects/items/miscellaneous.dm

```diff
@@ -179,7 +179,7 @@
 		return
 
 	to_chat(joe, "<span class='notice'>As you burn the picture, a nickname comes to mind...</span>")
-	var/nickname = input(joe, "Pick a nickname", "Mafioso Nicknames") as text|null
+	var/nickname = stripped_input(joe, "Pick a nickname", "Mafioso Nicknames", null, MAX_NAME_LEN, TRUE)
```

Comment:
```suggestion
	var/nickname = stripped_input(joe, "Pick a nickname", "Mafioso Nicknames", null, MAX_NAME_LEN - user.real_name.len, TRUE)
```

Reply:
web edit

Reply:
I'm not too keen on having it based on current name length since it can mean having a longer character name forces you out of having some nicknames when both could be acceptable.

I don't mind dividing it by 2 though with the idea this cap is meant to support a long First/Last name, and this would be a third "part" being tacked on to the name.